benchmarks/hex-simd/rust-benchmark/src/main.rs

```diff
@@ -0,0 +1,380 @@
+use sightglass_api as bench;
+use std::io::{self, Read};
+use std::str;
+
+#[cfg(target_arch = "x86")]
+use std::arch::x86::*;
+#[cfg(target_arch = "x86_64")]
+use std::arch::x86_64::*;
+
+fn main() {
+    let input = std::fs::read("./benchmark.wasm").unwrap();
+    let mut dst = vec![0; 2 * input.len()];
+    bench::start();
+    let s = hex_encode(&input, &mut dst).unwrap();
+    bench::end();
+    println!("{s}");
+}
+
+fn hex_encode<'a>(src: &[u8], dst: &'a mut [u8]) -> Result<&'a str, usize> {
+    let len = src.len().checked_mul(2).unwrap();
+    if dst.len() < len {
+        return Err(len);
+    }
+
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    {
+        if is_x86_feature_detected!("avx2") {
+            return unsafe { hex_encode_avx2(src, dst) };
+        }
+        if is_x86_feature_detected!("sse4.1") {
+            return unsafe { hex_encode_sse41(src, dst) };
+        }
+    }
+    #[cfg(target_arch = "wasm32")]
+    {
+        if true {
+            return unsafe { hex_encode_simd128(src, dst) };
+        }
+    }
+
+    hex_encode_fallback(src, dst)
+}
+
+#[target_feature(enable = "avx2")]
```

Comment:
Do we need to attribute this to another project?

Reply:
The original source [comes from here](https://github.com/rust-lang/stdarch/blob/master/examples/hex.rs) of which I believe I'm the main substantive author of. Should I copy in the LICENSE file here to be safe though?

Reply:
Heh, it's up to you I guess!